python/pyspark/ml/util.py

```diff
@@ -932,7 +932,7 @@ def loadMetadata(
             If non empty, this is checked against the loaded metadata.
         """
         metadataPath = os.path.join(path, "metadata")
-        spark = sc if isinstance(sc, SparkSession) else SparkSession._getActiveSessionOrCreate()
+        spark = cast(SparkSession, sc) if hasattr(sc, "createDataFrame") else SparkSession.active()
```

Comment:
`sc` can be:

- spark context
- classic spark session
- connect spark session

`isinstance(sc, SparkSession)` doesn't work for connect spark session, since we don't have a parent class for sessions